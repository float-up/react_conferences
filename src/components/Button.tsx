import React from "react";

interface IProps {
    text: string;
    disable?: boolean;
    onClick?: () => void;
}

export const Button: React.FC<IProps> = props => {

    const {
        text,
        onClick,
        disable
    } = props;

    return (
        <button className="waves-effect waves-light btn-large" disabled={disable} onClick={onClick}>{text}</button>
    )
}
