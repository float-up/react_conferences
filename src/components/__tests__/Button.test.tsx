import React from "react";
import {shallow} from "enzyme";
import {Button} from "../Button";

let onclick = jest.fn();
let wrapper = shallow(<Button text="Click me!" onClick={onclick} />)
describe('Button', () => {
    it('test Button render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('test props "text" in Button', () => {
        expect(wrapper.find('button').text()).toEqual('Click me!')
    })

    it('test onClick handler in Button', () => {
        wrapper.find('button').simulate('click', {target: {}})
        expect(onclick).toHaveBeenCalled();
    })
})
