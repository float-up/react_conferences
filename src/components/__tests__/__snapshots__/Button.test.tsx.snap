// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Button render 1`] = `
<button
  className="waves-effect waves-light btn-large"
  onClick={[MockFunction]}
>
  Click me!
</button>
`;
